runtime/common/src/migrations/liquidity_pools_v2.rs

```diff
@@ -0,0 +1,846 @@
+		*,
+	};
+
+	const LOG_PREFIX: &str = "InitMessageQueue";
+
+	pub struct Migration<T>(PhantomData<T>);
+
+	impl<T> OnRuntimeUpgrade for Migration<T>
+	where
+		T: pallet_liquidity_pools_gateway::Config<Message = Message>
+			+ frame_system::Config
+			+ pallet_liquidity_pools_gateway_queue::Config<
+				MessageNonce = u64,
+				Message = GatewayMessage<Message, RouterId>,
+			>,
+	{
+		fn on_runtime_upgrade() -> Weight {
+			let reads = 0u64;
+			let writes = 0u64;
+			OutboundMessageNonceStore::<T>::kill();
+
+			pallet_liquidity_pools_gateway_queue::MessageNonceStore::<T>::put(0u64);
+			Self::migrate_message_queue(reads, writes);
+			Self::migrate_failed_message_queue(reads, writes);
+			log::info!("{LOG_PREFIX}: Migration done!");
+
+			// Add weight from killing and setting nonce store
+			T::DbWeight::get().reads_writes(reads.saturating_add(1), writes.saturating_add(2))
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			assert!(OutboundMessageQueue::<T>::iter_values()
+				.into_iter()
+				.all(|(domain, _, _)| maybe_router_id(domain).is_some()));
+			assert!(FailedOutboundMessages::<T>::iter_values()
+				.into_iter()
+				.all(|(domain, _, _, _)| maybe_router_id(domain).is_some()));
+
+			let pending: u64 = OutboundMessageQueue::<T>::iter_keys()
+				.count()
+				.saturated_into();
+			let failed: u64 = FailedOutboundMessages::<T>::iter_keys()
+				.count()
+				.saturated_into();
+
+			log::info!("{LOG_PREFIX}: Pre checks done with {pending} items in OutboundMessageQueue and {failed} items in FailedOutboundMessages!");
+
+			Ok((pending, failed).encode())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(pre_state: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+			let (pre_pending, pre_failed): (u64, u64) = Decode::decode(&mut pre_state.as_slice())
+				.expect("pre_upgrade provides a valid state; qed");
+
+			let pending: u64 = pallet_liquidity_pools_gateway_queue::MessageQueue::<T>::iter_keys()
+				.count()
+				.saturated_into();
+			let failed: u64 =
+				pallet_liquidity_pools_gateway_queue::FailedMessageQueue::<T>::iter_keys()
+					.count()
+					.saturated_into();
+
+			assert_eq!(
+                pre_pending, pending,
+                "{LOG_PREFIX} POST: Mismatching number of pending messages in queue after migration!"
+            );
+			assert_eq!(
+                pre_failed, failed,
+                "{LOG_PREFIX} POST: Mismatching number of failed messages in queue after migration!"
+            );
+
+			log::info!("{LOG_PREFIX}: Post checks done!");
+
+			Ok(())
+		}
+	}
+
+	impl<T> Migration<T>
+	where
+		T: pallet_liquidity_pools_gateway::Config<Message = Message>
+			+ frame_system::Config
+			+ pallet_liquidity_pools_gateway_queue::Config<
+				MessageNonce = u64,
+				Message = GatewayMessage<Message, RouterId>,
+			>,
+	{
+		fn migrate_message_queue(mut reads: u64, mut writes: u64) {
```

Comment:
Shouldn't these reads/writes args be references?

Reply:
They could be yes. But is there any advantage to gain from that? Where do you see an issue with the current style?

Reply:
In the caller of this fn - https://github.com/centrifuge/centrifuge-chain/pull/1966/files#diff-701a3e1aba84712266444a4d0cdee431c7a71b7cd635255592de9e26f6d77714R252, if you don't provide mutable references they will remain 0?

Reply:
Fixed by [bf97ee8](https://github.com/centrifuge/centrifuge-chain/pull/1966/commits/bf97ee805ed145ed527fe03dd076b427e4e05234)